File: Algorithm/leetCode/two sum/JavaScript/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function(nums, target) {
  let [map, i, resultArr] = [{}, 0, []];
  const len = nums.length;
  for (; i < len; i++) {
  
    let tmp = target - nums[i];
    
    if (map[tmp] !== undefined) {
      resultArr.push(map[tmp]);
      resultArr.push(i);
      break;
    }
    map[nums[i]] = i;
  }

  return resultArr;
}

File: Algorithm/leetCode/two sum/c++/two-sum.cpp
/**
 * not good solution（空间复杂度太高了）
 */

class Solution {
  public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
      vector<int>tmp;
      for(int i = 0; i < nums.size()-1; i++){
        for(int j = i+1; j < nums.size(); j++){
          if(nums[i] + nums[j] == target){
            tmp.push_back(i);
            tmp.push_back(j);
            return tmp;
          }
        }
      }
      return tmp;
    }
};